Update slice parameter with additional information (#3350)
opensearch-project/documentation-website: _api-reference/document-apis/reindex.md

socket_timeout | The wait time for socket reads. Default is 30s.
connect_timeout | The wait time for remote connection timeouts. Default is 30s.
size | The number of documents to reindex.

> slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting slices to `auto` allows OpenSearch choose the number of slices to use. This will use one slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.

Reviewer: ```suggestion
slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting slices to `auto` allows OpenSearch to choose the number of slices to use. `auto` uses one slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.
```

Reviewer: Please confirm "This" refers to `auto`. Otherwise, please clarify what "This" refers to.

Reviewer 2: Should this be, "Setting _slice_ to "auto" allows..." 

You could combine the sentences so 'auto' is not the first word in the sentence: Setting this field to `auto` allows OpenSearch to control the number of slices to use, which is one slice per shard, up to a maximum of 20.

> slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting slices to `auto` allows OpenSearch choose the number of slices to use. This will use one slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.

Author: ```suggestion
slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting this field to `auto` allows OpenSearch to control the number of slices to use, which is 1 slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.
```
